### PR TITLE
Storage Autoclass samples

### PR DESCRIPTION
**Pull request by hemanshv**
Added  samples that demonstrate the following functionality:

- Enabling autoclass for a storage bucket
- Retrieving autoclass data of a storage bucket

### CODE REVIEW

**Comment by hemanshv** [comment]
Hi @jskeet and @amanda-tarafa ,
I have added samples for autoclass.
Please review and provide your valuable feedback.
Thanks,

File: storage/api/Storage.Samples/GetAutoclass.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+using Google.Cloud.Storage.V1;
+using static Google.Apis.Storage.v1.Data.Bucket;
+
+public class GetAutoclassSample
+{
+    public AutoclassData GetAutoclass(string bucketName)
+    {
+        var storage = StorageClient.Create();
+        var bucket = storage.GetBucket(bucketName);
+        return bucket.Autoclass;
```
Comment (cojenco) [inline storage/api/Storage.Samples/GetAutoclass.cs:26]: nit: consider adding a print statement in these 2 samples, as shown in the [canonical sample doc]([go/gcs-autoclass-canonical-sample](http://go/gcs-autoclass-canonical-sample)). It'd be nice to have this somewhat aligned across languages.
```
Console.WriteLine($"Autoclass enabled is set to {bucket.Autoclass.Enabled} for {bucketName} at {bucket.Autoclass.ToggleTime}.");
```

**Review by cojenco — COMMENTED** [review]
One nit, otherwise LGTM; Deferring to the C# experts for further review, thanks!

**Comment by cojenco** [comment]
Added a do not merge label as we're coordinating to merge/release in the week of Nov. 2

**Comment by cojenco** [comment]
> Added a do not merge label as we're coordinating to merge/release in the week of Nov. 2

Update - Autoclass GA is tentatively postponed to Nov 7 as the backend team is monitoring feature & UI rollout. I will comment once the timeline is confirmed. Thanks for your patience.

**Review by cojenco — APPROVED** [review]
API feature rollout is complete; ready to merge and release, thank you!

File: storage/api/Storage.Samples/SetAutoclass.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+using Google.Cloud.Storage.V1;
+using Google.Apis.Storage.v1.Data;
+using static Google.Apis.Storage.v1.Data.Bucket;
+using System;
+
+public class SetAutoclassSample
+{
+    public Bucket SetAutoclass(string bucketName, bool enable)
+    {
+        // Note: Only patch requests that disable autoclass are currently supported.
```
Comment (amanda-tarafa) [inline storage/api/Storage.Samples/SetAutoclass.cs:26]: Why receive the parameter then, that's very unclear for users. It'll be clearer if we do directly `Enabled = false`.

If patch ever supports enabling autoclas, we'll have to change this sample to remove this note, and we could add a parameter at that point. I'd prefer if we don't have a parameter now.

@cojenco any thoughts?

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]

File: storage/api/Storage.Samples/SetAutoclass.cs (hunk 2 above)
Comment (cojenco) [reply storage/api/Storage.Samples/SetAutoclass.cs:26]: Makes sense, let's remove the parameter for now. 

We were planning ahead for v2 that supports enablement on existing buckets. You're right that we'll have to update this sample anyhow at that point. I'll add a note in the tracking doc, thanks!

**Review by amanda-tarafa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**